### PR TITLE
feat(tailwindcss-transformer): Scaffold package

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

Initializes a new package `tailwindcss-transformer`. This is an experimental package we will be using to rebuild AIO components using Elements and Tailwind CSS.

↓ Input ↓

```jsx
export function Example({ flag }){
  let className = cn('absolute inset-0', flag && 'uppercase')
  return <div className={cn('flex items-center text-sm', className)} />
}
```

↓ Transform ↓

```jsx
export function Example({ flag }){
  let className = cn('cl-7601190e', flag && 'cl-d2cf63c7')
  return <div className={cn('cl-f64ae6a6', className)} />
}
```

```css
.cl-7601190e {
  @apply absolute inset-0;
}

.cl-d2cf63c7 {
  @apply uppercase;
}

.cl-f64ae6a6 {
  @apply flex items-center text-sm;
}
```

↓ Compile CSS ↓

```css
.cl-7601190e {
  position: absolute;
  inset: 0;
}

.cl-d2cf63c7 {
  text-transform: uppercase;
}

.cl-f64ae6a6 {
  display: flex;
  align-items: center;
  font-size: 0.875rem;
}
```

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/tailwindcss-transformer/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "email": "support@clerk.com",
+    "url": "git+https://github.com/clerk/javascript.git"
+  },
+  "scripts": {
+    "test": "vitest"
+  },
+  "dependencies": {
+    "@babel/parser": "^7.24.5",
+    "class-variance-authority": "^0.7.0",
+    "fast-glob": "^3.3.2",
```
Comment (kylemac) [inline packages/tailwindcss-transformer/package.json:17]: i dont think we use this one anymore

Comment (alexcarpenter) [reply packages/tailwindcss-transformer/package.json]: 👍🏼 https://github.com/clerk/javascript/pull/3397/commits/e454ff92945b4d512502bd8a7f975d52a2f04ae5

File: packages/tailwindcss-transformer/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2022 Clerk, Inc.
```
Comment (LekoArts) [inline packages/tailwindcss-transformer/LICENSE:3]:
```suggestion
Copyright (c) 2024 Clerk, Inc.
```

File: packages/tailwindcss-transformer/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<p align="center">
+  <a href="https://clerk.com?utm_source=github&utm_medium=clerk_remix" target="_blank" rel="noopener noreferrer">
```
Comment (LekoArts) [inline packages/tailwindcss-transformer/README.md:2]:
```suggestion
  <a href="https://clerk.com?utm_source=github&utm_medium=clerk_tailwindcss_transformer" target="_blank" rel="noopener noreferrer">
```

File: packages/tailwindcss-transformer/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  "scripts": {
+    "test": "vitest run --pool forks"
+  },
+  "dependencies": {
+    "@babel/parser": "^7.24.5",
+    "postcss": "^8.4.38",
+    "postcss-value-parser": "^4.2.0",
+    "recast": "^0.23.7",
+    "tailwindcss": "^3.4.3",
+    "vitest": "^1.6.0"
```
Comment (LekoArts) [inline packages/tailwindcss-transformer/package.json:20]: We're using Jest everywhere else so Vitest needs to be changed here to Jest

**Review by LekoArts — CHANGES_REQUESTED** [review]

File: packages/tailwindcss-transformer/package.json (hunk 4 above)
Comment (kylemac) [reply packages/tailwindcss-transformer/package.json:20]: @LekoArts i checked in with @BRKalow about vitest and his conclusion was this wasn't a problem. its probably worth surfacing as a topic with a few more folks before we make any changes here @alexcarpenter (particularly bc this is all a bit exploratory atm)

**Comment by alexcarpenter** [comment]
Closing in favor of the work done in https://github.com/clerk/javascript/pull/3402

**Pull request closed without merging**